### PR TITLE
[Review] Rewrite the Article of Style Guide

### PR DESCRIPTION
**Pull request by MegaApplePi**
- [x] Rewrites the Article of Style Guide from scratch.
- [x] Adjust the README.md file to list "Getting Started"
- [x] have a _higher up_ check this
#104

### CODE REVIEW

File: wiki/Article_Style_Guide/en.md [hunk 1]
```diff
@@ -1,192 +1,176 @@
...
-4.  Features that are not currently in the game should avoid having their own article.
-    1.  This excludes features which have been removed and April Fool's day features, which may be noted in articles affected by them.
+GFM has a small amount of markup that **anyone** can learn it in less than a day!
+However, don't expect a lot out of Markdown as it is intended to be used as a bare minimum markup language.
 
-### Community  
+### Syntax
 
-1.  Gameplay strategies, guides, how-tos, etc., should be subarticles of [Guides](/wiki/List_of_Guides).
+If you need help with GFM syntax, see [markdown syntax - GitHub Guides](https://guides.github.com/pdfs/markdown-cheatsheet-online.pdf) or [Markdown Cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet).
```
Comment (TPGPL) [inline wiki/Article_Style_Guide/en.md]: capitalize `markdown`

File: wiki/Article_Style_Guide/en.md [hunk 2]
```diff
@@ -1,192 +1,176 @@
...
 
-Article titles
---------------
+TOC (Table of Contents) are automatically generated on the osu!wiki.
+Because of this, there is no need to include one yourself.
 
-1.  Titles are to be written in sentence case. The first letter of a title must always be capitalised except in cases which are dully justified. Otherwise, words are to be capitalised only if they would be in running text.
-    1.  As *osu!* should never be capitalised, it shouldn't be capitalised in the article title either. *e.g.* [osu!mania](/wiki/Game_Modes/osu!mania). As the wiki forces the capitalisation on the first letter of the title, use `{{DISPLAYTITLE:osu!mania}}` to alter manually the display title of the article.
-        1.  Titles with a language namespace should be follow this format `{{DISPLAYTITLE:osu!mania (Language Name)}}`. For example, the [French osu!mania](/wiki/Game_Modes/osu!mania) article should use `{{DISPLAYTITLE:osu!mania (Français)}}`
+## Code Blocks
```
Comment (TPGPL) [inline wiki/Article_Style_Guide/en.md]: Code =/= Code Clocks

File: wiki/Article_Style_Guide/en.md [hunk 3]
```diff
@@ -1,192 +1,176 @@
...
+No.
 
+The osu!wiki uses GFM.
+GFM is Markdown which is basically a subset of HTML that contains these styles:
 
-*In osu! Standard, beatmaps are composed of three different gameplay elements: circles, sliders and spinners.*
+- **bold**
+- _italics_
+- ~~strikethrough~~
+- `code blocks`
```
Comment (TPGPL) [inline wiki/Article_Style_Guide/en.md]: same there

File: wiki/Article_Style_Guide/en.md [hunk 4]
```diff
@@ -1,192 +1,176 @@
...
-1.  Gameplay strategies, guides, how-tos, etc., should be subarticles of [Guides](/wiki/List_of_Guides).
+If you need help with GFM syntax, see [markdown syntax - GitHub Guides](https://guides.github.com/pdfs/markdown-cheatsheet-online.pdf) or [Markdown Cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet).
 
-### Wiki Rules  
+## Headings
 
-1.  Parody, comedic, nonsense, hoax, and speculation articles, or any other article that could mislead players are not allowed to be created.
-2.  Articles created for the purpose of advertising are not allowed.
-3.  Articles about fan communities are not allowed due to advertising issues.
+All articles **must have one level one heading** with the content being the article's title.
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]:
```Markdown
All articles **must have a level one heading of a translated article's title**.
```

File: wiki/Article_Style_Guide/en.md (hunk 1 above)
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: that is the name of the title of the page it links to

Comment (TPGPL) [inline wiki/Article_Style_Guide/en.md]: [markdown syntax - GitHub Guides] 
I mean this one

File: wiki/Article_Style_Guide/en.md [hunk 5]
```diff
@@ -1,192 +1,176 @@
...
-3.  Previous article title.
-    1.  An exception is if the previous title was not commonly used.
+## Article Titles
 
-4.  Alternate capitalisation or form, including changing the title to plural case.
-5.  A part of a merged or multi-topic article, such as a "Double Time" for [Game Modifiers](/wiki/Game_Modifiers).
+The title of an article is the name of the folder that the article.
+Replace the underscores (`_`) with spaces (` `).
+For English titles, if you need to reword the title, you must rename the folder, with rare expections.
+Translated articles of an article will be located inside the appropriate English-named folder.
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: Honestly speaking, I have no idea what you are saying now.

Put an example here. Like (deliberately done badly):-

```Markdown
Article_Style_Guide (filename)

# Panduan Gaya Artikel (example in ml.md)
```

File: wiki/Article_Style_Guide/en.md [hunk 6]
```diff
@@ -1,192 +1,176 @@
...
-3.  It is unnecessary to include an acronym or an abbreviation in addition to the name in a title. *e.g.* Double Time (DT).
-4.  Definite and indefinite articles (the, a, and an) should not be placed at the beginning of titles unless they are part of a proper name.
+Use code blocks when noting:
 
-Writing
--------
+1. a key on the keyboard
+   - i.e. `B`, `Alt` or `Control` + `Shift` + `A`
+2. a name of a button
+   - i.e. `Let's do it.` or `Noooo! I didn't mean to!`
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: Nitpicking.
```Markdown
   - i.e. `1. Let's do it.` or `2. Noooo! I didn't mean to!`
```

File: wiki/Article_Style_Guide/en.md [hunk 7]
```diff
@@ -1,192 +1,176 @@
...
 
-2.  Article titles should be in the singular form to maintain consistency. *e.g.* Beatmap and not Beatmaps.
-3.  It is unnecessary to include an acronym or an abbreviation in addition to the name in a title. *e.g.* Double Time (DT).
-4.  Definite and indefinite articles (the, a, and an) should not be placed at the beginning of titles unless they are part of a proper name.
+Use code blocks when noting:
 
-Writing
--------
+1. a key on the keyboard
+   - i.e. `B`, `Alt` or `Control` + `Shift` + `A`
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: Control -> Ctrl ?

File: wiki/Article_Style_Guide/en.md [hunk 8]
```diff
@@ -1,192 +1,176 @@
...
+3. must not specify the specific file name **regardless of language** (osu!wiki will handle this)
+  - do: `[Glossary](/wiki/Glossary/)`
+  - don't: `[Glossary](/wiki/Glossary/fr.md)`
+4. must not link to section headings (section headings on osu!wiki does not work)
 
-Pages on the wiki should only use *British English* as a matter of consistency. For instance, “color” should be “colour”, “center” should be “centre”, “skillful” should be “skilful”, “paralleled” should be “parallelled”, and “analyze” should be “analyse”. It is recommended to get accustomed to British English and American English spelling differences before writing an article.
+External links must follow these rules:
 
-### Capitalisation
+- prefer the `https` protocol, if available
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: You do not really need to specify the https. Just

//osu.ppy.sh

will work, amazingly (See the Best_Of).

File: wiki/Article_Style_Guide/en.md (hunk 5 above)
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: i.e. list of guides of en.md, zh.md, pl.md are located in the `List_of_Guides`

File: wiki/Article_Style_Guide/en.md (hunk 6 above)
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: Nitpicking
``` `4. Noooo! I didn't mean to! ````

File: wiki/Article_Style_Guide/en.md (hunk 8 above)
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: but that was what @TPGPL says not to do .-.

File: wiki/Article_Style_Guide/en.md [hunk 9]
```diff
@@ -1,192 +1,280 @@
...
+This guide is to serve as a way to help keep consistency in clarity, formatting, and layout between all articles of the osu!wiki.
 
-This article aims to provide a comprehensive style guide for all osu!wiki articles to follow. Although Wikipedia already provides a more general [style guide](https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style), a more specific one is necessary for *osu!* specific guidelines. As such, only guidelines pertaining to the osu!wiki and its basic formatting rules should be included here.
+All articles should use plain English.
+Your word choice should explain such things in layman's terms (basically, ask yourself, "If the reader is reading this, would they have to lookup any of the words?").
 
-Notability
-----------
+All articles must have proper grammar, correct spelling, and correct information.
+Try your best to do all of those in your pull request and know that reviewers will look for these things before sending it off to the live osu!wiki.
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L9: Consider Revising to a more passive format.

File: wiki/Article_Style_Guide/en.md [hunk 10]
```diff
@@ -1,192 +1,280 @@
...
-1.  Parody, comedic, nonsense, hoax, and speculation articles, or any other article that could mislead players are not allowed to be created.
-2.  Articles created for the purpose of advertising are not allowed.
-3.  Articles about fan communities are not allowed due to advertising issues.
+Translated articles are to be placed in the appropriate English-named folder.
 
-Articles in the "User:" namespace are exempt from the notability guidelines. They can be used for anything, provided they follow the other Wiki Rules.
+For a list of the 2 lettered language codes, see [List of ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) (the 639-1 column of the table).
 
-### Redirects
+## Headings
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L32 : Provide a sample for the succeding sentence

File: wiki/Article_Style_Guide/en.md [hunk 11]
```diff
@@ -1,192 +1,280 @@
...
-    -   Images should not have the sole purpose of showcasing a bug, instead report the bug in the [Help subforum](https://osu.ppy.sh/forum/5).
-    -   Images showcasing usage of specific features for decoration should be avoided.
--   When adding screenshots to an article, make sure the screenshots use the default skin. Screenshots that use custom skins or other custom content are not allowed.
-    -   This does not apply to articles covering skinning.
-    -   In the case of Taiko screenshots (Gameplay only), the Taiko skin can be used instead
--   Articles should only have one image showcasing an individual attribute of the articles content. For example, a slider in the beatmap editor.
--   Image captions should not have periods at the end, unless the phrase is a full sentence.
--   Images should showcase the most up to date version of *osu!* available for the content.
-    -   Images that are outdated are subject to be removed.
+## FAQ
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L234 : This should be linking the FAQ on Getting Started instead (and extend that) IMHO

File: wiki/Article_Style_Guide/en.md [hunk 12]
```diff
@@ -1,192 +1,280 @@
...
+- `skilful` instead of `skillful`
+- `analyse` instead of `analyze`
 
 ### Capitalisation
 
-1.  The name of the game *osu!* should never be capitalised.
-    1.  Words following the exclamation mark of *osu!*, if not divided by a blank space, should not be capitalised, *e.g.* osu!mania. If they are divided by a blank space, they must be capitalised if they are proper nouns, *e.g.* osu! Standard.
+#### osu!
+
+The name of the game `osu!` is **never** capitalised, even if it is the first word in the sentence.
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L89 : Include "capitalized" in bolding

File: wiki/Article_Style_Guide/en.md [hunk 13]
```diff
@@ -1,192 +1,280 @@
...
+
+- `osu!wiki`
+- `osu!alumni`
+- `osu!direct`
+
+#### Language Names
+
+When referring the name of a language, capitalize the first letter of that language.
+
+- The **Maltese** chat channel are for those who speak **Maltese**.
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L103: was there a Maltese chat channel? :thinking:

File: wiki/Article_Style_Guide/en.md [hunk 14]
```diff
@@ -1,192 +1,280 @@
...
+The osu!wiki will give every image a max-width, being the width of the article body.
 
-Example:
+Names of images must be somewhat meaningful, please don't use the timestamp or long hashes for media file names.
 
+Images that are used in multiple different articles must go to the `/wiki/shared` folder.
+Images that are used for one article are to be placed inside their folders for linking.
 
-*In osu! Standard, beatmaps are composed of three different gameplay elements: circles, sliders and spinners.*
+It is suggusted to add a subfolder for the images, i.e. `/img/` for all images.
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L205 : adding a ``/`` before the dirname would resolve to root, right?

File: wiki/Article_Style_Guide/en.md [hunk 15]
```diff
@@ -1,192 +1,280 @@
...
+If you have thumbnails and full sized images, it is suggested to use two folders and name the files the same; i.e. `/pr/` for preview and `/fs/` for full size.
 
-### Section headings
+Thumnails are suggested to have a width of 160px and be in the `.jpg` format (unless it has transparency).
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L208 : Typo. Change to "Thumbnails"

File: wiki/Article_Style_Guide/en.md (hunk 9 above)
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: hmm... what do you mean? Can you give me an example of the "more passive format"?

File: wiki/Article_Style_Guide/en.md (hunk 11 above)
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: well, this is a FAQ of MediaWiki and GitHub stuff, not really pertaining to osu! (the game and itself)

File: wiki/Article_Style_Guide/en.md (hunk 13 above)
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: I guess, I'll use an actual channel, `#Spanish` channel

File: wiki/Article_Style_Guide/en.md [hunk 16]
```diff
@@ -1,192 +1,318 @@
...
+All section headings are to use title case, just like the article titles.
+
+Section headings must **never** have a link in it.
+Instead place the link underneath the section heading.
+For example:
+
+```
+## Editor
+
+For a full explaination, see [Editor](/wiki/Editor/).
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: Nitpicking ze link.

`For a full explanation, see [Beatmap_Editor](/wiki/Beatmap_Editor/).`

File: wiki/Article_Style_Guide/en.md [hunk 17]
```diff
@@ -1,192 +1,318 @@
...
+```
+
+## TOC
+
+The TOC (Table of Contents) are automatically generated on the osu!wiki.
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: `The TOC (Table of Contents) will be automatically generated on the osu!wiki.`

File: wiki/Article_Style_Guide/en.md [hunk 18]
```diff
@@ -1,192 +1,318 @@
...
+- prefer the `https://` protocol
+- if referencing another site, must link to a reputable source
+- must be a clean and direct link
+  - do not use a third-party shortened link
+  - do not use links that link to ads
+  - do not use a url with long url queries, parameters, or fragments
+- the link name must be the title text of the page it is linking to
 
-Game modifiers as well as their abbreviations *should* always be capitalised.
+Do use use the word `here` as the link text.
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: Do use use?

```Markdown
Do not use the word `here` as the link text.
```

File: wiki/Article_Style_Guide/en.md [hunk 19]
```diff
@@ -1,192 +1,318 @@
...
+As it may seem convenient to use, it could be misleading.
+Instead, take the page's title (from the top of page, or from the `<title>` tag) and use that as the link text.
 
-Examples:
+You can use either the reference or inline style links.
+If you are using the reference style linking, it is sugguested to place the reference links at the top of the article for quicker access.
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: For link maintenance.
Pretty sure I will be _pissed_ if I have to scroll down to the end of the page just to fix the links.

YMMV so no comment.

File: wiki/Article_Style_Guide/en.md [hunk 20]
```diff
@@ -1,192 +1,318 @@
...
+```
 
-*Circle Size, known as "CS" in Edit's song selection, determines the size of circles and sliders*
+All beatmapset difficutly urls looks like this: `osu.ppy.sh/s/{BeatmapSetID}`
 
-<!-- -->
+## Images
 
-#### Gameplay Elements  
+**All image file extensions _MUST_ use lower cased letters!!**
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: Spacing Nitpicks

```Markdown
**All image file extensions _MUST_ use lowercase letters!!**
```

File: wiki/Article_Style_Guide/en.md (hunk 9 above)
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: I will try.

```Markdown
Reviewers may ask for changes in your pull request for blunders or suggestions.
A good osu!wiki writer should read the reviews to improve the quality of writing to ensure for the best for an osu!wiki reader.
```

...Yeah, I went _a bit too_ overboard on that. No idea is that passive enough or too idealistic.

File: README.md [hunk 21]
```diff
@@ -1,11 +1,12 @@
...
 [osu! on the web](https://osu.ppy.sh) | [dev chat](https://discord.gg/ppy)
 
-future home of osu! wiki
+future home of osu!wiki
 
 # Requirements
 
-A markdown compatible editor.
+1. A GitHub account
+2. A general understanding of the feature-branch workflow
```
Comment (RaikireHiuduo) [inline README.md]:
```Markdown
1. A [GitHub account](https://help.github.com/articles/signing-up-for-a-new-github-account/ "GitHub Help in Signing up an account").
2. A general understanding of the feature-branch workflow
3. Knowledge in [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet "adam-p's cheatsheet for Markdown"), specifically [GitHub-Flavoured Markdown(GFM)](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/ "GitHub Help on Writing on GitHub").
4. Able to write in English, preferably British English.
```

File: wiki/Article_Style_Guide/en.md [hunk 22]
```diff
@@ -1,192 +1,343 @@
...
+## Miscellaneous
 
-### Italics
+- Do not use an inline image link for osu!supporter
+  - use `[osu!supporters](/wiki/osu!supporter/)`, not `[![](/wiki/shared/osu!support.png/)](/wiki/osu!supporter/)`
+- These words are spelt as follows (note the space):
+  - `hit circles`
+  - `approach circles`
+  - `slider ticks`
+  - `slider ball`
```
Comment (RaikireHiuduo) [inline wiki/Article_Style_Guide/en.md]: Add:-
```Markdown
- `slider path`
```

**Comment by RaikireHiuduo** [comment]
Currently, when referring to the playable songs, I am used to saying

What I used | What it meant and should be
---|---
Difficulty, or Beatmap's difficulty | Beatmap/Map
Beatmap/Map | Beatmapset/Mapset

Each beatmaps in a beatmapset was usually named as difficulty names, so I ended up with a tendency to refer a beatmap as beatmap's difficulty and a beatmapset as a beatmap instead.

I think this is more to Glossary material but it is still related to the ASG so I post here.

If you just want me to enforce it blindly and expecting the reader to actually bother to read the (whole, or finding that specific term manually in) Glossary or search the term, sure.

---

Oh, and some more stuffs on spacing (in event of `to space or not to space`, err on the side of to space).

```Markdown
- hit burst
- game mode
- game mod
- play style
- hit sound (?)
```

File: README.md [hunk 23]
```diff
@@ -1,11 +1,14 @@
 # osu-wiki
 [osu! on the web](https://osu.ppy.sh) | [dev chat](https://discord.gg/ppy)
 
-future home of osu! wiki
+future home of osu!wiki
 
 # Requirements
```
Comment (TPGPL) [inline README.md]: i would change it into "##"

File: README.md [hunk 24]
```diff
@@ -1,11 +1,14 @@
...
 
-A markdown compatible editor.
+1. A [GitHub account](https://help.github.com/articles/signing-up-for-a-new-github-account/ "GitHub Help in Signing up an account").
+2. A general understanding of the feature-branch workflow.
+3. Knowledge in [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet "adam-p's cheatsheet for Markdown"), specifically [GitHub-Flavoured Markdown(GFM)](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/ "GitHub Help on Writing on GitHub").
+4. Able to write in English, preferably British English.
 
 # Licence
```
Comment (TPGPL) [inline README.md]: same

File: README.md [hunk 25]
```diff
@@ -14,3 +17,11 @@ The majority of content in this repository is licensed under [CC-BY-NC 4.0](http
 As this is a wiki, there may be content with third party licences. These licences will be cited local to the content, and override the global licence file.
 
 Please note that this *does not cover* the usage of the "osu!" or "ppy" branding in any software, resources, advertising or promotion, as this is protected by trademark law. If you require clearance for the use of these terms, please [contact us](mailto:contact@ppy.sh).
+
+# Getting Started
```
Comment (TPGPL) [inline README.md]: same

File: wiki/Article_Style_Guide/en.md [hunk 26]
```diff
@@ -1,192 +1,369 @@
...
 
-This article aims to provide a comprehensive style guide for all osu!wiki articles to follow. Although Wikipedia already provides a more general [style guide](https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style), a more specific one is necessary for *osu!* specific guidelines. As such, only guidelines pertaining to the osu!wiki and its basic formatting rules should be included here.
+All articles should use plain English.
+Your word choice should explain such things in layman's terms (basically ask yourself, "If the reader is reading this, would they have to lookup any of the words?").
 
-Notability
-----------
+All articles must have proper grammar, correct spelling, and correct information.
+Know that reviewers may ask for changes in your pull request for blunders or suggestions.
+A good osu!wiki writer should read these reviews to help improve the overall quality of these articles to ensure for the best for an osu!wiki reader.
```
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: L10 : Revise. Suggested :  ``A good osu!wiki writer should read these reviews to help improve the overall quality of these articles to ensure the most optimal experience for an osu!wiki reader``.

File: README.md [hunk 27]
```diff
@@ -1,16 +1,27 @@
 # osu-wiki
 [osu! on the web](https://osu.ppy.sh) | [dev chat](https://discord.gg/ppy)
 
-future home of osu! wiki
+future home of osu!wiki
```
Comment (sr229) [inline README.md]: L4: Revise to ``The future home of osu!wiki``

Comment (MegaApplePi) [inline README.md]: I wouldn't want to modify the readme file any further but alright

File: wiki/Article_Style_Guide/en.md (hunk 11 above)
Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: I was pertaining to #102 
😂

Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: and #123; I mentioned it to RaikireHiuduo in a review comment for that pr. Would it fit better in the how to contribute to the osu!wiki article?

Comment (sr229) [inline wiki/Article_Style_Guide/en.md]: Yes.

File: wiki/Article_Style_Guide/en.md [hunk 28]
```diff
@@ -1,192 +1,311 @@
...
+
+- `osu!wiki`
+- `osu!alumni`
+- `osu!direct`
+
+#### Language Names
+
+When referring the name of a language, capitalize the first letter of that language.
+
+- The **#Spanish** chat channel are for those who speak **Spanish**.
```
Comment (Repflez) [inline wiki/Article_Style_Guide/en.md]: Chat names are always in lower case.
